### PR TITLE
Reduce synchronization on CqlClientFactory

### PR DESCRIPTION
**Pull request by taikhuu**
CqlClientFactory have synchronized eveything method
which will greatly reduce parallelism unnecessary.
Move synchronized to initalize methods.

### CODE REVIEW

File: src/main/java/com/hmsonline/trident/cql/CqlClientFactory.java [hunk 1]
```diff
@@ -13,29 +13,43 @@
...
     abstract Cluster.Builder getClusterBuilder();
 
-    public synchronized Session getSession(String keyspace) {
+    public Session getSession(String keyspace) {
         Session session = sessions.get(keyspace);
         if (session == null) {
             LOG.debug("Constructing session for keyspace [" + keyspace + "]");
-            session = getCluster().connect(keyspace);
+            session = initializeSession(keyspace);
             sessions.put(keyspace, session);
```
Comment (baontq) [inline src/main/java/com/hmsonline/trident/cql/CqlClientFactory.java]: It doesn't hurt, but it's confusing to put session to the map again.

**Comment by baontq** [comment]
LGTM

**Comment by bflad** [comment]
I'll cut a release tomorrow morning unless there are any blocker comments.
Thanks!

File: src/main/java/com/hmsonline/trident/cql/CqlClientFactory.java (hunk 1 above)
Comment (taikhuu) [inline src/main/java/com/hmsonline/trident/cql/CqlClientFactory.java]: i will correct this

**Comment by baontq** [comment]
LGTM. Thanks @taikhuu @bflad

**Comment by pabrahamsson** [comment]
@baontq @taikhuu @bflad 
https://oss.sonatype.org/content/repositories/releases/com/hmsonline/storm-cassandra-cql/0.3.2/storm-cassandra-cql-0.3.2.jar

**Pull request merged**